test/factory/smartWalletFactory.test.ts

```diff
@@ -1,497 +1,557 @@
-//                     callForwarder: factory.address
-//                 }
-//             );
-
-//             const { signature, suffixData } = signRequest(
-//                 owner.privateKey,
-//                 relayRequest,
-//                 chainId
-//             );
-
-//             await expect(
-//                 factory.relayedUserSmartWalletCreation(
-//                     relayRequest.request,
-//                     suffixData,
-//                     worker,
-//                     signature,
-//                     {
-//                         from: worker
-//                     }
-//                 )
-//             ).to.be.rejectedWith(
-//                 'Returned error: VM Exception while processing transaction: revert nonce mismatch'
-//             );
-//         });
-
-//         it('Should fail when signature does not match', async () => {
-//             const relayRequest = createRequest(
-//                 {
-//                     from: owner.address,
-//                     to: logicAddress,
-//                     data: initParams,
-//                     tokenContract: token.address,
-//                     tokenAmount: '5000',
-//                     tokenGas: '0',
-//                     index: index,
-//                     relayHub: worker
-//                 },
-//                 {
-//                     callForwarder: factory.address
-//                 }
-//             );
-
-//             const { signature, suffixData } = signRequest(
-//                 owner.privateKey,
-//                 relayRequest,
-//                 chainId
-//             );
-
-//             relayRequest.request.from = otherAccount;
-
-//             await expect(
-//                 factory.relayedUserSmartWalletCreation(
-//                     relayRequest.request,
-//                     suffixData,
-//                     worker,
-//                     signature,
-//                     {
-//                         from: worker
-//                     }
-//                 )
-//             ).to.be.rejectedWith(
-//                 'Returned error: VM Exception while processing transaction: revert Signature mismatch'
-//             );
-//         });
-//     });
-// });
+import { expect } from 'chai';
+import { SmartWallet, SmartWalletFactory, UtilToken } from 'typechain-types';
+import { ethers } from 'hardhat';
+import { constants, utils, Wallet } from 'ethers';
+import { createValidPersonalSignSignature } from '../utils/createValidPersonalSignSignature';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { createDeployRequest } from './utils';
+import {
+  getLocalEip712DeploySignature,
+  TypedDeployRequestData,
+} from '../utils/EIP712Utils';
+import { getSuffixData, HARDHAT_CHAIN_ID } from '../smartwallet/utils';
+
+const minIndex = 0;
+const maxIndex = 1000000000;
+
+const nextIndex = () =>
+  Math.floor(Math.random() * (maxIndex - minIndex + 1) + minIndex);
```

Comment:
I don't like the fact that the next index depends completely on a random number, hence the tests are not repeatable. If possible, I'd avoid it or replace it with a [random generator that accepts a seed](https://github.com/davidbau/seedrandom).

Reply:
I can use apply this, but then I think that needs to be applied to the rif-relay testing as-well.

Reply:
This is the only change that is pending, I don't see whats the issue about depending on a random number, if we don't have any type of dependency on this, during the tests.

Reply:
updated